test/unit/math/prim/fun/offset_multiplier_transform_test.cpp

```diff
@@ -65,9 +68,542 @@ TEST(prob_transform, offset_multiplier_f) {
+  {
+    auto result
+        = stan::math::offset_multiplier_constrain(x_vec, offsetd, sigmad);
+    for (size_t i = 0; i < result.size(); ++i) {
+      for (size_t j = 0; j < result[i].size(); ++j) {
+        EXPECT_FLOAT_EQ(result[i](j), stan::math::offset_multiplier_constrain(
+                                          x_vec[i](j), offsetd, sigmad));
+      }
+    }
+    auto x_vec_free
+        = stan::math::offset_multiplier_free(result, offsetd, sigmad);
+    for (size_t i = 0; i < x_vec.size(); ++i) {
+      for (size_t j = 0; j < x_vec[i].size(); ++j) {
+        EXPECT_FLOAT_EQ(x_vec[i].coeff(j), x_vec_free[i].coeff(j));
+      }
+    }
+  }
+}
+
+TEST(prob_transform, offset_multiplier_free_exception) {
+  double xd = 1.0;
+  Eigen::VectorXd x(2);
+  x << 1.0, 2.0;
+
+  double mu = 2.0;
+
+  double mu_bad1 = stan::math::INFTY;
+  Eigen::VectorXd mu_bad2(2);
+  mu_bad2 << stan::math::INFTY, 2.0;
+  Eigen::VectorXd mu_bad3(3);
+
+  double sigma = 2.0;
+
+  double sigma_bad1 = stan::math::INFTY;
+  double sigma_bad2 = 0.0;
+  double sigma_bad3 = -2.0;
+  Eigen::VectorXd sigma_bad4(2);
+  sigma_bad4 << stan::math::INFTY, 2.0;
+  Eigen::VectorXd sigma_bad5(2);
+  sigma_bad5 << -1.0, 2.0;
+  Eigen::VectorXd sigma_bad6(3);
+
+  EXPECT_THROW(stan::math::offset_multiplier_free(xd, mu_bad1, sigma),
+               std::domain_error);
+
+  EXPECT_THROW(stan::math::offset_multiplier_free(xd, mu, sigma_bad1),
+               std::domain_error);
+  EXPECT_THROW(stan::math::offset_multiplier_free(xd, mu, sigma_bad2),
+               std::domain_error);
+  EXPECT_THROW(stan::math::offset_multiplier_free(xd, mu, sigma_bad3),
+               std::domain_error);
+
+  EXPECT_THROW(stan::math::offset_multiplier_free(x, mu, sigma_bad1),
+               std::domain_error);
+  EXPECT_THROW(stan::math::offset_multiplier_free(x, mu, sigma_bad2),
+               std::domain_error);
+  EXPECT_THROW(stan::math::offset_multiplier_free(x, mu, sigma_bad3),
+               std::domain_error);
+  EXPECT_THROW(stan::math::offset_multiplier_free(x, mu, sigma_bad4),
+               std::domain_error);
+  EXPECT_THROW(stan::math::offset_multiplier_free(x, mu, sigma_bad5),
+               std::domain_error);
+  EXPECT_THROW(stan::math::offset_multiplier_free(x, mu, sigma_bad6),
+               std::invalid_argument);
+
+  EXPECT_THROW(stan::math::offset_multiplier_free(x, mu_bad1, sigma),
+               std::domain_error);
+  EXPECT_THROW(stan::math::offset_multiplier_free(x, mu_bad2, sigma),
+               std::domain_error);
+  EXPECT_THROW(stan::math::offset_multiplier_free(x, mu_bad3, sigma),
+               std::invalid_argument);
+}
+
+TEST(prob_transform, offset_multiplier_consistent_sizes) {
+  Eigen::VectorXd x(4);
+  x << 1.0, 2.0, 3.0, 4.0;
+
+  double mu = 2.0;
+
+  Eigen::MatrixXd sigma(2, 2);
+  sigma << 1.0, 2.0, 3.0, 4.0;
+
+  EXPECT_THROW(stan::math::offset_multiplier_free(x, mu, sigma),
+               std::invalid_argument);
```

Comment:
I think everything is here except this. `check_consistent_sizes` makes sure that all arguments are either of the same size or scalar. We need something that checks that all arguments are of the same dimensions or scalar. I'd be fine changing the behavior of `check_consistent_sizes` since I think thus far we're only using it with vectors and it seems fine to expand the use to matrices.

Anyway it needs to be here cause this will blow up with an Eigen dimension mismatch error.

Also since fma is an exposed function ([here](https://mc-stan.org/docs/2_26/functions-reference/composed-functions.html)) we need size checks on it (or users will get Eigen errors when things are mismatches sizes).

All of this applies to lb/ub/lub_constrain as well I think.

Reply:
This is fixed now.

Reply:
Also lb/ub/lub_constrain handle things correctly already.